### ISSUE TITLE
Wrong implementation of heat index: stair steps around HI = 26.1 C.

### ISSUE THREAD

**Issue by darrenfreeman** [dolezsa/thermal_comfort#341]
I am seeing stair-step glitches in the `heat_index` sensor, for example with 72% humidity, and temperature going from 25.5 C to 25.7 C. The resulting `heat_index` jumps drastically.

See the attached chart, with `heat_index` crossing my temperature setpoint unreasonably often - it happens to be 26.5 C.

The implementation differs significantly from the formula given in the source material from NOAA, and it can easily explain this issue. I am testing a solution, and will create a PR in the next day or two.

![2023-12-13-heat-index-glitches](https://github.com/dolezsa/thermal_comfort/assets/4440028/c010804c-ca8a-4fa4-9fe2-054fda9b8d78)

**Comment by higbyk** [comment]
Agreed, the steps are part of the design, and knowing it will cause intermittent jumps, you'll have to use a different formula or account for it by smoothing or a delay (over 79°F for 2 minutes).

You might consider making a template sensor (can be done through the UI now) with a smoothed version of this logic.

**Comment by puetsch** [comment]
In case anyone wants to use it, I've created these macros and then made a template sensor. The result is a heat index that is a continuous function. According to the wikipedia page, it should be great for temperatures between 70 and 115 °F (21–46 °C), so could get funky under 20°C or so.

```jinja
{% macro to_fahrenheit(t_c) -%}
  {{ t_c * 9/5 + 32 }}
{%- endmacro %}

{% macro to_celsius(t_f) -%}
  {{ (t_f - 32) * 5/9 }}
{%- endmacro %}

{# Formula from https://en.wikipedia.org/wiki/Heat_index#Formula #}
{% macro heat_index(sensor_tc, sensor_rh) -%}

  {% set t_c = states(sensor_tc)|float %}
  {% set rh = states(sensor_rh)|float %}
  
  {% set t_f = to_fahrenheit(t_c)|float %}
  
  {% set c1 = 0.363445176 %}
  {% set c2 = 0.988622465 %}
  {% set c3 = 4.777114035 %}
  {% set c4 = -0.114037667 %} 
  {% set c5 = -8.50208E-4 %}
  {% set c6 = -2.0716198E-2 %}
  {% set c7 = 6.87678E-4 %}
  {% set c8 = 2.74954E-4 %}
  
  {% set hi_f = c1 + c2*t_f + c3*rh %}
  {% set hi_f = hi_f + c4*t_f*rh + c5*t_f*t_f + c6*rh*rh %}
  {% set hi_f = hi_f + c7*t_f*t_f*rh + c8*t_f*rh*rh %}
  
  {% set hi_c = to_celsius(hi_f) %}
  
  {{ hi_c }}
{%- endmacro %}
```
